functions/functionsv2/imagemagick/imagemagick.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START functions_imagemagick_setup]
+
+// Package imagemagick contains an example of using ImageMagick to process a
+// file uploaded to Cloud Storage.
+package imagemagick
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+
+	"cloud.google.com/go/storage"
+	vision "cloud.google.com/go/vision/apiv1"
+	"github.com/GoogleCloudPlatform/functions-framework-go/functions"
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	visionpb "google.golang.org/genproto/googleapis/cloud/vision/v1"
+)
+
+// Global API clients used across function invocations.
+var (
+	storageClient *storage.Client
+	visionClient  *vision.ImageAnnotatorClient
+)
+
+func init() {
+	// Declare a separate err variable to avoid shadowing the client variables.
+	var err error
+
+	bgctx := context.Background()
+	storageClient, err = storage.NewClient(bgctx)
+	if err != nil {
+		log.Fatalf("storage.NewClient: %v", err)
+	}
+
+	visionClient, err = vision.NewImageAnnotatorClient(bgctx)
+	if err != nil {
+		log.Fatalf("vision.NewAnnotatorClient: %v", err)
+	}
+	functions.CloudEvent("blur-offensive-images", blurOffensiveImages)
+}
+
+// [END functions_imagemagick_setup]
+
+// [START functions_imagemagick_analyze]
+
+// GCSEvent is the payload of a GCS event.
+// additional fields are documented at
+// https://cloud.google.com/storage/docs/json_api/v1/objects#resource
+type GCSEvent struct {
+	Bucket string `json:"bucket"`
+	Name   string `json:"name"`
+}
+
+// blurOffensiveImages blurs offensive images uploaded to GCS.
+func blurOffensiveImages(ctx context.Context, e cloudevents.Event) error {
+	outputBucket := os.Getenv("BLURRED_BUCKET_NAME")
+	if outputBucket == "" {
+		return errors.New("environment variable BLURRED_BUCKET_NAME must be set")
+	}
+
+	gcsEvent := &GCSEvent{}
+	if err := e.DataAs(gcsEvent); err != nil {
+		return fmt.Errorf("DataAs: failed to decode event data: %v", err)
```

Comment:
nit: `e.DataAs` would match go convetion to start with a lowercase letter with errors.